src/legacy/core_plugins/input_control_vis/public/components/vis/range_control.js

```diff
@@ -17,175 +17,93 @@
-      formatedValue = value.toFixed(decimalPlaces);
-    }
-    return formatedValue;
-  };
+    return null;
+  }
+
+  onChangeComplete = _.debounce(value => {
+    const controlValue = {
+      min: value[0],
+      max: value[1]
+    };
+    this.props.stageFilter(this.props.controlIndex, controlValue);
+  }, 200);
+
+
 
   renderControl() {
     if (!this.props.control.isEnabled()) {
       return (
-        <EuiRange
+        <ValidatedDualRange
           disabled
+          showInput
         />
       );
     }
 
+    const decimalPlaces = _.get(this.props, 'control.options.decimalPlaces', 0);
+    const min = floorWithPrecision(this.props.control.min, decimalPlaces);
+    const max = ceilWithPrecision(this.props.control.max, decimalPlaces);
+
+    const ticks = [
+      { value: min, label: min },
+      { value: max, label: max }
+    ];
+
     return (
-      <EuiFormRow
-        isInvalid={!this.state.isRangeValid}
-        error={this.state.errorMessage ? [this.state.errorMessage] : []}
-        data-test-subj="rangeControlFormRow"
-      >
-        <EuiFlexGroup alignItems="center" gutterSize="s">
-          <EuiFlexItem grow={false}>
-            <input
-              id={`${this.props.control.id}_min`}
-              name="min"
-              type="number"
-              data-test-subj="rangeControlMinInputValue"
-              className="euiFieldNumber"
-              value={this.state.minValue}
-              min={this.props.control.min}
-              max={this.props.control.max}
-              onChange={this.handleMinChange}
-            />
-          </EuiFlexItem>
-          <EuiFlexItem grow={false} className="icvInputRange__container">
-            <InputRange
-              maxValue={this.props.control.max}
-              minValue={this.props.control.min}
-              step={this.props.control.options.step}
-              value={this.state.sliderValue}
-              onChange={this.handleOnChange}
-              onChangeComplete={this.handleOnChangeComplete}
-              draggableTrack={true}
-              ariaLabelledby={this.props.control.id}
-              formatLabel={this.formatLabel}
-            />
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <input
-              id={`${this.props.control.id}_max`}
-              name="max"
-              type="number"
-              className="euiFieldNumber"
-              data-test-subj="rangeControlMaxInputValue"
-              value={this.state.maxValue}
-              min={this.props.control.min}
-              max={this.props.control.max}
-              onChange={this.handleMaxChange}
-            />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiFormRow>
+      <ValidatedDualRange
+        id={this.props.control.id}
+        min={min}
+        max={max}
+        value={this.state.value}
+        onChange={this.onChangeComplete}
+        showInput
+        showRange
+        showTicks
+        ticks={ticks}
```

Comment:
So the main issue that was happening was with the ticks. I see that you're only adding the min and max as ticks, which isn't necessary since we have the `showLabels` prop which automatically appends the min and max values passed. Like so:

<img width="306" alt="screen shot 2019-02-15 at 16 29 41 pm" src="https://user-images.githubusercontent.com/549577/52885357-edde1a00-313e-11e9-9fa9-af076674cd9f.png">

However, I was going to suggest auto-filling the inputs with the min/max numbers so the range wasn't in a weird state like the above, but basically looked like it has the whole range selected. Therefore, if we do that, then we don't need the `showLabels` because the inputs will already show those numbers and it's quite easy to reset  to the whole range with the reset button:

<img width="447" alt="screen shot 2019-02-15 at 16 33 13 pm" src="https://user-images.githubusercontent.com/549577/52885484-680e9e80-313f-11e9-8598-a30645e343b4.png">

Reply:
I've do have to put in a fix for the input sizing and I saw a few other bugs, but this should allow you to continue with the PR.

Reply:
> I see that you're only adding the min and max as ticks, which isn't necessary since we have the `showLabels` prop which automatically appends the min and max values passed

The problem is `showLabels` is that it displays the labels inline with the slider. For input controls, horizontal space is at a premium  so I wanted the labels below the slider to save horizontal space. I also really like the ability to select a tick to set the value

> However, I was going to suggest auto-filling the inputs with the min/max numbers so the range wasn't in a weird state like the above

You can not pre set the values to the entire range because then that suggests that there is a range filter already applied. The default state is that the slider has no values. Then you can use the slider to specify a range.

Reply:
Then we need to somehow disconnect empty values from the look of the range because the range looks broken.

Also the ticks are broken right now because custom ticks don't correctly take into account the min value, it's always based on `0` being the min. I'm looking into it now.